lib/ansible/modules/cloud/amazon/ec2_vpc_subnet.py

```diff
@@ -45,15 +52,39 @@
     choices: [ 'present', 'absent' ]
   vpc_id:
     description:
-      - "VPC ID of the VPC in which to create the subnet."
-    required: false
+      - "VPC ID of the VPC in which to create or delete the subnet."
+    required: true
     default: null
   map_public:
     description:
-     - "Specify true to indicate that instances launched into the subnet should be assigned public IP address by default."
+      - "Specify true to indicate that instances launched into the subnet should be assigned public IP address by default."
     required: false
     default: false
     version_added: "2.4"
+  assign_ipv6_on_create:
```

Comment:
I'm not big on this name, since it's not clear on creation of what. Maybe `assign_instances_ipv6` or similar would be clearer?